nectar/src/command/trade/event_loop.rs

```diff
@@ -211,17 +198,33 @@ impl EventLoop {
 
         self.maker.free_funds(dai, btc);
 
-        let _ = self
+        let peer_db_res = self
             .database
             .remove_active_peer(&finished_swap.peer)
             .await
-            .map_err(|error| tracing::error!("Unable to remove from active takers: {}", error));
+            .context("Unable to remove from active takers");
 
-        let _ = self
+        let swap_db_res = self
             .database
             .remove_swap(&swap_id)
             .await
-            .map_err(|error| tracing::error!("Unable to delete swap from db: {}", error));
+            .context("Unable to delete swap from db");
+
+        if let Err(history_err) = history_res {
+            Err(history_err.context(format!(
+                "Issue inserting in history; peer db: {:?}; swap db: {:?}",
+                peer_db_res, swap_db_res
+            )))
```

Comment:
Nit, if you use `format!` with `context`, you might want to get in the habit of using `with_context` to avoid the allocation if it is not actually executed (i.e. in the Ok) case :)